pkg/network/tracer/connection/kprobe/tracer.go

```diff
@@ -147,13 +147,6 @@ func New(config *config.Config, constants []manager.ConstantEditor) (connection.
 }
 
 func (t *kprobeTracer) Start(closeFilter func(*network.ConnectionStats) bool) (err error) {
-	// this is to ensure that it is safe to call Stop() on a kprobeTracer that fails to Start()
-	defer func() {
-		if err != nil {
-			t = nil
```

Comment:
may leave the defer func here but with t.Stop() and we could remove the other t.Stop() call before the return

Reply:
Yeah, that's a good idea 👍